WordPress/Classes/Models/Post.m

```diff
@@ -4,16 +4,31 @@
 #import "Coordinate.h"
 #import "NSMutableDictionary+Helpers.h"
 #import "ContextManager.h"
+#import <WordPress-iOS-Shared/NSString+XMLExtensions.h>
+
+@interface Post()
+@property (nonatomic, strong) NSString *storedContentPreviewForDisplay;
+@end
 
 @implementation Post
 
-@dynamic geolocation, tags, postFormat;
-@dynamic latitudeID, longitudeID, publicID;
+@dynamic geolocation;
+@dynamic tags;
+@dynamic postFormat;
+@dynamic latitudeID;
+@dynamic longitudeID;
+@dynamic publicID;
 @dynamic categories;
 @synthesize specialType;
+@synthesize storedContentPreviewForDisplay;
 
 #pragma mark - NSManagedObject subclass methods
 
+- (void)awakeFromFetch
+{
+    [self buildContentPreview];
```

Comment:
Minor comment: Please, also invoke [super awakeFromFetch]. Not sure what the super method actually does, but [it's required in the docs](https://developer.apple.com/library/ios/documentation/Cocoa/Reference/CoreDataFramework/Classes/NSManagedObject_Class/index.html#//apple_ref/occ/instm/NSManagedObject/awakeFromFetch).